api/api.go

```diff
@@ -91,6 +92,10 @@ const (
 	SpecSharedv4ServiceName  = "sharedv4_svc_name"
 	SpecFastpath             = "fastpath"
 	SpecAutoFstrim           = "auto_fstrim"
+	SpecBackendType          = "backend"
```

Comment:
Is the idea that for other vendors/"backend" types, they'd add their own `SpecBackend<Vendor><Type>` constant?

Reply:
yes this could be potentially used for the other type of backend in the future, now we use `backend` key but in the code we translate it to `ProxyProtocol`, and save the specs for new backend into `ProxySpec` as discussed offline